libs/audit-service/src/main/java/uk/gov/di/ipv/core/library/auditing/helpers/AuditExtensionsHelper.java

```diff
@@ -1,130 +1,126 @@
 
-        var drivingPermit = credentialSubject.path(VC_DRIVING_PERMIT);
-        if (drivingPermit.isArray() && !drivingPermit.isEmpty()) {
-            var docExpiryDate = drivingPermit.path(0).path(VC_EXPIRY_DATE).asText();
-            return new AuditRestrictedF2F(name, docExpiryDate);
-        }
+        if (vc.getCredential() instanceof RiskAssessmentCredential riskAssessmentCredential) {
+            var riskAssessments = riskAssessmentCredential.getEvidence();
 
-        var brp = credentialSubject.path(VC_RESIDENCE_PERMIT);
-        if (brp.isArray() && !brp.isEmpty()) {
-            var docExpiryDate = brp.path(0).path(VC_EXPIRY_DATE).asText();
-            return new AuditRestrictedF2F(name, docExpiryDate);
+            return new AuditExtensionsVcEvidence<>(
+                    issuer, riskAssessments, isSuccessful, vot, isUkIssued, age);
         }
 
-        var idCard = credentialSubject.path(VC_ID_CARD);
-        if (idCard.isArray() && !idCard.isEmpty()) {
-            var docExpiryDate = idCard.get(0).path(VC_EXPIRY_DATE).asText();
-            return new AuditRestrictedF2F(name, docExpiryDate);
-        }
+        return new AuditExtensionsVcEvidence<>(
+                jwtClaimsSet.getIssuer(), null, isSuccessful, vot, isUkIssued, age);
+    }
 
-        return new AuditRestrictedF2F(name);
+    public static AuditRestrictedF2F getRestrictedAuditDataForF2F(VerifiableCredential vc)
+            throws HttpResponseExceptionWithErrorBody {
+
+        if (vc.getCredential() instanceof IdentityCheckCredential identityCheckCredential) {
+            var credentialSubject = getCredentialSubjectOrThrow(identityCheckCredential);
+
+            var name = credentialSubject.getName();
+
+            var passport = credentialSubject.getPassport();
+            if (!isNullOrEmpty(passport)) {
+                var docExpiryDate = passport.get(0).getExpiryDate();
+                return new AuditRestrictedF2F(name, docExpiryDate);
+            }
+
+            var drivingPermit = credentialSubject.getDrivingPermit();
+            if (!isNullOrEmpty(drivingPermit)) {
+                var docExpiryDate = drivingPermit.get(0).getExpiryDate();
+                return new AuditRestrictedF2F(name, docExpiryDate);
+            }
+
+            var brp = credentialSubject.getResidencePermit();
+            if (!isNullOrEmpty(brp)) {
+                var docExpiryDate = brp.get(0).getExpiryDate();
+                return new AuditRestrictedF2F(name, docExpiryDate);
+            }
+
+            var idCard = credentialSubject.getIdCard();
+            if (!isNullOrEmpty(idCard)) {
+                var docExpiryDate = idCard.get(0).getExpiryDate();
+                return new AuditRestrictedF2F(name, docExpiryDate);
+            }
+
+            return new AuditRestrictedF2F(name);
+        } else {
+            LOGGER.error(LogHelper.buildLogMessage("VC not of type IdentityCheckCredential."));
+            return new AuditRestrictedF2F(null);
```

Comment:
Returning an empty `AuditRestrictedF2F` here instead of throwing an error when it's not an IdentityCheckType since it didn't throw an error before if name was null